catalogue/webapp/pages/works.js

```diff
@@ -427,8 +440,19 @@ const Works = ({ works }: Props) => {
 
 WorksSearchProvider.getInitialProps = async (ctx: Context): Promise<Props> => {
   const query = ctx.query.query;
+  const _dateFrom = makeApiSafeDateString(ctx.query._dateFrom);
+  const _dateTo = makeApiSafeDateString(ctx.query._dateTo);
   const page = ctx.query.page ? parseInt(ctx.query.page, 10) : 1;
 
+  // API doesn't seem to like dates above the year 9999
```

Comment:
It does feel weird touch what is relatively stupid bits of functionality with this? Could we handle this closer to the side-effect i.e. in the component? If someone passes a bad date, this will error silently, which doesn't feel right?

Reply:
> It does feel weird touch what is relatively stupid bits of functionality with this?

I don't understand.

If someone hacks the URL to a year over 9999 (unlikely, I know), is there another way to prevent us 500ing?

Reply:
We 400 - which is completely acceptable and expected no? Rather be told what's wrong over failing silently?

Reply:
You're right – I was seeing an error coming from the analytics script which threw me off.